geo-types/src/vertex_string.rs

```diff
@@ -0,0 +1,674 @@
+            }
+        }
+
+        let need_update = match self.closest_neighbor {
+            Some((_, c)) => c > cost,
+            None => true,
+        };
+
+        if need_update {
+            self.closest_neighbor.replace((index, cost));
+        }
+    }
+}
+
+impl<T> PartialEq for  Vertex<T>
+where
+    T: Float
+{
+    fn eq(&self, other: &Self) -> bool {
+        self.coordinate == other.coordinate
+    }
+}
+
+/// An undirected graph data structure that can be used to build more complex geo concepts,
+/// e.g. a geo map. The struct contains a collection of connected or lone vertices, and each vertex
+/// has its own struct to express its relationship with neighboring connected vertices.
+///
+/// The `VertexString` can be built by default from a `Line` iterator or collection container. But
+/// it can also be converted from most common `Line`-based data structure, such as a vector of the `Line`s,
+/// a `LineString`, a vector of `LineString`s, or a `MultiLineString`.
+///
+/// # Examples
+///
+/// Create a `VertexString` from a vector of `Line`s, where each line represent an edge
+///
+/// ```
+/// use geo_types::{Line, VertexString};
+///
+/// let vec: Vec<Line<f32>> = vec![
+///     Line::from([(10., 5.), (15., 10.)]),
+///     Line::from([(15., 10.), (20., 15.)]),
+///     Line::from([(20., 15.), (10., 5.)]),
+/// ];
+///
+/// let graph = VertexString::from(vec);
+/// //let mut it = graph.vertex_iter();
+///
+/// assert_eq!(it.next().unwrap().get_coordinate(), (10., 5.).into());
+//  assert_eq!(it.next().unwrap().get_coordinate(), (15., 10.).into());
+/// assert_eq!(it.next().unwrap().get_coordinate(), (20., 15.).into());
+/// assert_eq!(it.next(), None);
+/// ```
+///
+#[derive(Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct VertexString<T: CoordinateType>
+where
+    T: Float
+{
+    // The store of the vertices
+    vector: Vec<Vertex<T>>,
+
+    // The index store that is used to help retrieve a vertex fast if only given the vertex coordinate
+    index: HashMap<String, usize>,
+
+    // The type of algorithm used to determine the edge cost (i.e. the distance between two vertices)
+    cost_calc_type: Cost<T>,
+}
+
+impl<T> VertexString<T>
+where
+    T: Float
+{
+    /// Create a `VertexString` from the iterator of the edges connecting the vertices. If a vertex
+    /// is not connected to any other vertices in the graph, then the start/end coordinate of the
+    /// edge (aka the `Line<T>` object) shall be identical.
+    pub fn new<C>(src: C) -> Self
+    where
+        C: Iterator<Item = Line<T>>
+    {
+        VertexString::build(src, None)
+    }
+
+    /// Create a `VertexString` from the iterator of the edges connecting the vertices.
+    /// If the size of the vertex coordinates container is known, using this API will
+    /// help improve the overall performance.
+    pub fn new_with_size<C>(src: C, size: usize) -> Self
```

Comment:
this should be called `with_size`. see the 'General constructors' section [here](https://rust-lang-nursery.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case)

Reply:
I'll update the constructor name to match the linked instructions.